torchtune/modules/loss/loss_protocols.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Protocol
+
+import torch
+
+
+class SFTLinearLoss(Protocol):
+    """Protocol for loss functions in torchtune used in Supervised Finetune recipes that require
+    model output linear projection weights in loss computation."""
+
+    linear_loss: bool = True
```

Comment:
Can we leave a comment in these classes explaining what this field means? Also an example usage in the docstrings of both would help a lot imo